Mathlib/CategoryTheory/Generator.lean

```diff
@@ -567,4 +569,98 @@ theorem wellPowered_of_isDetector [HasPullbacks C] (G : C) (hG : IsDetector G) :
   haveI := small_subsingleton ({G} : Set C)
   wellPowered_of_isDetecting hG
 
+theorem wellPowered_of_isSeparator [HasPullbacks C] [Balanced C] (G : C) (hG : IsSeparator G) :
+    WellPowered C := wellPowered_of_isDetecting hG.isDetector
+
+section HasGenerator
+
+variable (E : Type u₃) [Category.{v₃} E]
+
+/--
+A category instantiates `HasSeparator` if and only if it has a separator.
+See also `IsSeparator`.
+-/
+class HasSeparator : Prop :=
```

Comment:
This syntax is [going away soon](https://github.com/leanprover/lean4/pull/5542).
```suggestion
class HasSeparator : Prop where
```